### PR TITLE
Deploy account util

### LINKED ISSUE #333: Deploy account util
we'd be so happy if we turned 

```python
my_account = await starknet.deploy(
    "contracts/Account.cairo",
    constructor_calldata=[my_signer.public_key]
)
```

into

```python
my_account = await Account.deploy(my_signer.public_key)
```

Not sure if it makes sense to have this here or Nile tbh.

### PR DESCRIPTION
**Pull request by andrew-fleming**
This PR creates a pattern for abstracting Account contract deployments by utilizing the `global` keyword to initialize the StarkNet state.

### `State` class
- wrapper for `Starknet.empty()`
- sets `global starknet`
- includes the method `init()` to initialize

### `Account` class
- wrapper for Account contract deployment
- invoking `await State.init()` gives `Account` (and other contract deployments) access to the initialized `starknet` state
- includes
  - the method `deploy` for deployments
  - the variable `get_class` to return the Account contract class

Resolves #333.

### CODE REVIEW

**Comment by andrew-fleming** [comment]
Hmm tests took 52 min here 🤔

File: docs/Utilities.md [hunk 1]
```diff
@@ -22,7 +22,9 @@ The following documentation provides context, reasoning, and examples for method
 * [Memoization](#memoization)
   * [`get_contract_def`](#get_contract_def)
   * [`cached_contract`](#cached_contract)
-* [Signer](#signer)
+* [State](#state)
+* [Account](#account)
+* [TestSigner](#testsigner)
```
Comment (JulissaDantes) [inline docs/Utilities.md:27]: Instead of TestSigner is MockSigner now

File: docs/Utilities.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ The following documentation provides context, reasoning, and examples for method
 * [Memoization](#memoization)
   * [`get_contract_class`](#get_contract_class)
   * [`cached_contract`](#cached_contract)
+* [State](#state)
```
Comment (martriay) [inline docs/Utilities.md:26]: this will need to be adapted to the new doc system and format

**Review by martriay — COMMENTED** [review]
Great addition and simplification! Is `State` only meant to be used next to accounts? Otherwise we're missing a few `starknet.empty()` in the tests for EthAccount, ERC165, Initializable, Reentrancy, and SafeMath.

**Comment by andrew-fleming** [comment]
> Great addition and simplification!

Thank you so much!

> Is `State` only meant to be used next to accounts? Otherwise we're missing a few `starknet.empty()` in the tests for EthAccount, ERC165, Initializable, Reentrancy, and SafeMath.

Ah good catch. The `State` object can be used everywhere not just with accounts. I'll update the mentioned tests for consistency

**Review by martriay — APPROVED** [review]
This is ready, would you mind updating the docs to the new system?

**Comment by andrew-fleming** [comment]
> This is ready, would you mind updating the docs to the new system?

On it!

**Review by martriay — APPROVED** [review]

**Pull request merged**